sympy/core/_print_helpers.py

```diff
@@ -0,0 +1,47 @@
+"""
+Base class to provide str and repr hooks that `init_printing` can overwrite.
+
+This is exposed publicly in the `printing.defaults` module,
+but cannot be defined there without causing circular imports.
+"""
+from typing import Any
+
+
+class Printable:
+    """
+    The default implementation of printing for SymPy classes.
+
+    This implements a hack that allows us to print elements of built-in
+    Python containers in a readable way. Natively Python uses ``repr()``
+    even if ``str()`` was explicitly requested. Mix in this trait into
```

Comment:
> Natively Python uses ``repr()``
    even if ``str()`` was explicitly requested.

I don't understand what this sentence is saying. How does this change the behavior of Python calling repr when the user calls str? Is this referring to the behavior where Python uses `repr` on `str(list)`, or something else?

Reply:
I guess this was from the old class. This is probably what was meant:

```py
>>> class Test:
...     def __repr__(self):
...         return 'repr'
...     def __str__(self):
...         return 'str'
>>> str([Test()])
'[repr]'
```

It's the reasoning why we keep `repr` and `str` the same for SymPy types.

Reply:
Yeah, this comment is unchanged, it just moved file.